src/main/java/gift/controller/AdminProductController.java

```diff
@@ -0,0 +1,92 @@
+package gift.controller;
+
+import gift.dto.ProductRequestDto;
+import gift.dto.ProductResponseDto;
+import gift.service.ProductService;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.validation.BindingResult;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+
+@Controller
+@RequestMapping("/admin/products")
+public class AdminProductController {
+
+    private final ProductService productService;
+
+    public AdminProductController(ProductService productService) {
+        this.productService = productService;
+    }
+
+    //목록 화면
+    @GetMapping
+    public String list(@RequestParam(defaultValue = "0") int page,
+                       @RequestParam(defaultValue = "10") int size,
+                       Model model) {
+
+        // 서비스 호출
+        List<ProductResponseDto> products = productService.getProductList(page, size);
+
+        // 총 페이지 계산용
+        int total = productService.getProductList(0, Integer.MAX_VALUE).size();
+        int totalPages = (int) Math.ceil((double) total / size);
+
+        // 뷰로 전달
+        model.addAttribute("products", products);
+        model.addAttribute("currentPage", page);
+        model.addAttribute("size", size);
+        model.addAttribute("totalPages", totalPages);
+
+        return "admin/products";            // templates/admin/products.html
+    }
+
+    //등록 화면
+    @GetMapping("/new")
+    public String newForm(Model model) {
+        model.addAttribute("form", new ProductRequestDto());
+        return "admin/product-form";
+    }
+
+    @PostMapping("/new")
+    public String createProduct(@ModelAttribute("form") ProductRequestDto requestDto,
+                                BindingResult bindingResult) {
+        if (bindingResult.hasErrors()) {
+            return "admin/product-form";
+        }
+
+        productService.createProduct(requestDto);
+        return "redirect:/admin/products";
+    }
+
+
+    //수정 화면
+    @GetMapping("/{id}/edit")
+    public String editForm(@PathVariable Long id, Model model) {
+        ProductResponseDto responseDto = productService.getProduct(id);
+
+        ProductRequestDto requestDto = new ProductRequestDto();
+        requestDto.setName(responseDto.getName());
+        requestDto.setPrice(responseDto.getPrice());
+        requestDto.setImageUrl(responseDto.getImageUrl());
+
+        model.addAttribute("form", requestDto);
```

Comment:
혹시 responseDto를 바로 넣지 않고 requestDto로 변환해서 넣는 이유가 있으신가요?

Reply:
product-form.html는 값을 수정, 등록 해서 서버로 다시 보내기 위한 화면입니다. 그래서 이미 유효성 검사도 하고 있고 수정하고 등록하는 용도로 만든 productRequestDto 가 더 적합하다고 생각했습니다 . ProductResponseDto에는 setter가 필요할 것 같지 않아 만들지 않았는데 form에 바인딩 하려면 setter가 원래 있는 RequestDto를 쓰는게 편할 것 같았습니다

제가 생각한 점 외에 RequestDto보다는 ResponseDto를 써야 로직이 좋아진다면 코멘트 부탁드립니다. 코멘트에 대해 고민하고 공부한뒤 변경하겠습니다 !

Reply:
```suggestion
        ProductResponseDto responseDto = productService.getProduct(id);

        model.addAttribute("form", responseDto);
```
그냥 responseDto를 넣으면 문제가 생길까 싶었어요 👀  
그리고 client 에게 돌려주는 상황인데 requestDto가 넘어간다는 것도 어색한 감이 있어서 comment 달았습니다! responseDto에도 setter를 추가해서 사용해도 되지 않을까요?